nevergrad/functions/test_base.py

```diff
@@ -23,7 +23,13 @@ def test_experiment_function() -> None:
         constkwarg="blublu",
         plop=p.Choice([3, 4]),
     ))
+    iarrayfunc = base.ArrayExperimentFunction(_arg_return,   # type: ignore
+        p.Array(shape=(10,)),
+        symmetry=247,
+    )
     np.testing.assert_equal(ifunc.dimension, 8)
+    np.testing.assert_equal(iarrayfunc.dimension, 10)
+    iarrayfunc(np.zeros(10))
```

Comment:
you should check that some coefficients were indeed changed, and that it handles boundary. An independent tests would be better.

Reply:
Also, I suppose iarrayfunc.copy() should return the same func, does that work?

Reply:
Tests added, fixes applied, thx for your help.

Reply:
(now adding cases with box-constraints)